src/lib/features/chain_of_trust.js

```diff
@@ -25,6 +26,8 @@ class ChainOfTrust {
     this.hash = crypto.createHash('sha256');
     let armoredKey = fs.readFileSync(task.runtime.signingKeyLocation, 'ascii');
     this.key = openpgp.key.readArmored(armoredKey).keys;
+    let sk = Buffer.from(fs.readFileSync(task.runtime.ed25519SigningKeyLocation, 'ascii'), 'base64');
+    this.ed25519Key = Buffer.from(fs.readFileSync(task.runtime.ed25519SigningKeyLocation, 'ascii'), 'base64');
```

Comment:
Here I think it is better to use the `async` version of these function, to not harm concurrency. The best way is to use promises, like so:
```javascript
let sk = Buffer.from(await new Promise((accept, reject) => 
  fs.readFile('path', (err, data) => err ? reject(err) : accept(data))), 'base64');
```

Reply:
The `ascii` argument comes before the callback :)

Reply:
Thanks! Fixed. I think I've addressed all of your concerns now.